dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyClientHandler.java

```diff
@@ -78,27 +76,35 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
         }
     }
 
-
     @Override
     public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
         super.write(ctx, msg, promise);
-        NettyChannel channel = NettyChannel.getOrAddChannel(ctx.channel(), url, handler);
-        try {
-            // if error happens from write, mock a BAD_REQUEST response so that invoker can return immediately without
-            // waiting until timeout. FIXME: not sure if this is the right approach, but exceptionCaught doesn't work
-            // as expected.
-            if (promise.cause() != null && msg instanceof Request) {
+        final NettyChannel channel = NettyChannel.getOrAddChannel(ctx.channel(), url, handler);
+        final boolean isMsg = msg instanceof Request;
```

Comment:
would it be better to name it 'isRequest'?

Reply:
my bad
I will fix it. Thanks for your suggestion.